core/src/main/java/com/google/errorprone/bugpatterns/RestrictedApiChecker.java

```diff
@@ -99,19 +99,29 @@ public Description matchNewClass(NewClassTree tree, VisitorState state) {
   }
 
   private Description checkMethodUse(ExpressionTree tree, VisitorState state) {
```

Comment:
I think this is only going to check `@RestrictedApi` on types if a method is called on those types. If we add support for this I think we'd want to disallow all uses of `@RestrictedApi`-annotated types, e.g. if they appear as qualifiers for fields, or as class literals, etc. instead of just when they appear as the qualifier of a method invocation.